presentation/testing/src/test/IModelUtilities.test.ts

```diff
@@ -0,0 +1,174 @@
+    sinon.restore();
+  });
+
+  const setupSnapshot = () => {
+    snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
+    sinon.stub(SnapshotDb, "createEmpty").returns(snapshotDbMock.object);
+    sinon.stub(SnapshotConnection, "openFile");
+  };
+
+  it("insertModel calls iModel.models.insertModel", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    modelPropsMock = moq.Mock.ofType<ModelProps>();
+    modelsMock = moq.Mock.ofType<IModelDb.Models>();
+    imodelMock.setup((x) => x.models).returns(() => modelsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertModel(modelPropsMock.object);
+
+    modelsMock.verify(async (x) => x.insertModel(modelPropsMock.object), moq.Times.once());
+  });
+
+  it("insertElement calls iModel.models.insertElement", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    elementPropsMock = moq.Mock.ofType<ElementProps>();
+    elementsMock = moq.Mock.ofType<IModelDb.Elements>();
+    imodelMock.setup((x) => x.elements).returns(() => elementsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertElement(elementPropsMock.object);
+
+    elementsMock.verify(async (x) => x.insertElement(elementPropsMock.object), moq.Times.once());
+  });
+
+  it("insertAspect calls iModel.models.insertAspect", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    elementAspectPropsMock = moq.Mock.ofType<ElementAspectProps>();
+    elementsMock = moq.Mock.ofType<IModelDb.Elements>();
+    imodelMock.setup((x) => x.elements).returns(() => elementsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertAspect(elementAspectPropsMock.object);
+
+    elementsMock.verify(async (x) => x.insertAspect(elementAspectPropsMock.object), moq.Times.once());
+  });
+
+  it("createCode calls iModel.codeSpecs.getByName", () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    codeSpecsMock = moq.Mock.ofType<CodeSpecs>();
+    codeScopePropsMock = moq.Mock.ofType<CodeScopeProps>();
+    codeSpecMock = moq.Mock.ofType<CodeSpec>();
+    codeSpecsMock.setup((x) => x.getByName(BisCodeSpec.drawing)).returns(() => codeSpecMock.object).verifiable(moq.Times.once());
+    imodelMock.setup((x) => x.codeSpecs).returns(() => codeSpecsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.createCode(codeScopePropsMock.object, BisCodeSpec.drawing, "codeValue");
+
+    codeSpecsMock.verifyAll();
+  });
+
+  it("calls IModelJsFs.mkdirSync if directory does not exist", async () => {
+    setupSnapshot();
+    sinon.stub(IModelJsFs, "existsSync").returns(false);
+    const mkdirFake = sinon.fake();
+    const unlinkFake = sinon.fake();
+    sinon.replace(IModelJsFs, "mkdirSync", mkdirFake);
+    sinon.replace(IModelJsFs, "unlinkSync", unlinkFake);
+
+    await buildTestIModel("name", () => {});
+
+    expect(mkdirFake.calledOnce);
+    expect(unlinkFake.notCalled);
```

Comment:
A test should only test things it says it tests. In this case it says it's testing the directory creation when it doesn't exist, so it shouldn't be checking `unlink` call. 

Also, we should be checking the arguments. Firstly, we want to know they're correct. Secondly, it helps readability by showing what the test expects to be created / unlinked. E.g. in this case, without looking at the library source code, I don't know whether you're testing `unlink` for the directory or the file.

Reply:
Created two more tests for this. Also checking args now.